web/src/settings_notifications.js

```diff
@@ -121,18 +121,31 @@ export function set_enable_marketing_emails_visibility() {
 export function set_up(settings_panel) {
     const $container = $(settings_panel.container);
     const settings_object = settings_panel.settings_object;
-    const $notification_sound_elem = $(settings_panel.notification_sound_elem);
+    const $notification_sound_elem = $("#realm-default-notification-sound-audio");
     const for_realm_settings = settings_panel.for_realm_settings;
+    const $notification_sound_dropdown = $container.find(".setting_notification_sound");
+    const $discard_button = $container.find(".save-discard-widget-button.discard-button");
+
+    $discard_button.on("click", () => {
+        notifications.update_notification_sound_source($notification_sound_elem, settings_object);
+    });
+
+    $notification_sound_dropdown.on("change", () => {
+        const sound = $notification_sound_dropdown.val().toLowerCase();
+        notifications.update_notification_sound_source($notification_sound_elem, {
+            notification_sound: sound,
+        });
+    });
```

Comment:
It would probably be better to have this change handler code in `settings_realm_user_settings_defaults.js` since we only want this in default user settings. For personal settings, the sound source element is updated after receiving the event since there is no save-discard widget there.

Reply:
You can directly use the selector of the dropdown element to get the value selected in dropdown.

Reply:
I meant that you only need to move the below code to `settings_realm_user_settings_defaults.set_up` and there is no check for "none" value here.
```js
 $notification_sound_dropdown.on("change", () => {
        const sound = $notification_sound_dropdown.val().toLowerCase();
        notifications.update_notification_sound_source($notification_sound_elem, {
            notification_sound: sound,
        });
    });
```

Reply:
OK, I have done that and reviewed the previous bug in personal settings and it is resolved.